### DISCUSSION TITLE
groupBy ts error: Type '[]' is not assignable to type '[any, ...any[]]'.

### DISCUSSION THREAD

**Discussion by likezero** [DefinitelyTyped/DefinitelyTyped#56021] [Issues with a @types package]
after upgrade the latest version : @types/lodash: 4.14.175

this code report an error: Type '[]' is not assignable to type '[any, ...any[]]'.
```js
const groupRet = _.groupBy(firstFormatRet, o => o.productTag);
// more code

groupRet[category] = [];  // Type '[]' is not assignable to type '[any, ...any[]]'.
```

[a comment by an automated account is not shown]

**Reply by Louis-Tian** [reply]
I think this is related to https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55914.
`groupBy` will logically never return an empty array as value for any key. 
The inferred type `[T, ...T[]]` enforces this.
